packages/miniflare/src/workers/core/entry.worker.ts

```diff
@@ -208,11 +209,11 @@ export default <ExportedHandler<Env>>{
     const isProxy = request.headers.get(CoreHeaders.OP) !== null;
     if (isProxy) return handleProxy(request, env);
 
-    // `dispatchFetch()` will always inject the passed URL as a header. When
+    // `dispatchFetch()` will always inject this header. When
     // calling this function, we never want to display the pretty-error page.
     // Instead, we propagate the error and reject the returned `Promise`.
-    const isDispatchFetch =
-      request.headers.get(CoreHeaders.ORIGINAL_URL) !== null;
+    const disablePrettyErrorPage =
+      request.headers.get(CoreHeaders.DISABLE_PRETTY_ERROR) !== null;
```

Comment:
There was no example to follow with boolean headers so I used the existing check, as is.

Another options is:
```suggestion
    const enablePrettyErrorPage =
      Boolean(request.headers.get(CoreHeaders.PRETTY_ERROR) ?? 'true');
```

where `CoreHeaders.PRETTY_ERROR === "MF-Pretty-Error"` and `dispatchFetch` would set it with `forward.headers.set(CoreHeaders.PRETTY_ERROR, "false");`